docs/source/deduplication.rst

```diff
@@ -0,0 +1,40 @@
+DeDuplication of EBU-TT-Live documents
+======================================
+
+When documents are ReSequenced, duplication of ``style`` and ``region`` elements
+and attributes can occur. To address this, the DeDuplicator node takes in a
+``manifest_file``, and processes the document(s) with the
+:py:func:`ebu_tt_live.node.deduplicator.DeDuplicatorNode.remove_duplication` function.
+
+After copying ``styling`` and ``layout`` into a ``list()`` and setting them up for new
+``style`` and ``region`` elements, respectively, the ``list()`` is passed through the
+:py:func:`ebu_tt_live.node.deduplicator.DeDuplicatorNode.CollateUniqueVals` function.
+
+Because ``style`` and ``region`` elements can have ``style`` attributes, these
+are deduplicated first. Each element is then passed through the
+:py:class:`ebu_tt_live.node.deduplicator.ComparableElement` class, which processes
+each attribute, omitting the ``xml:id`` and using the
+:py:func:`ebu_tt_live.node.deduplicator.ReplaceNone` function to replace empty
+attributes with a non-legal character to avoid collisions, and assigns a hash
+to each element. The hash is then stored in ``old_id_dict`` as a key-value pair,
+where the ``xml:id`` is the ``key`` and the hash is the ``value``. The hash is also stored
+in ``hash_dict`` where the hash is the ``key``, and the ``value`` is the contents of the element.
+
+The ``list()`` is then passed through the :py:func:`ebu_tt_live.node.deduplicator.DeDuplicatorNode.AppendNewElements`
+function, which takes in the list of elements, the path to the parent element
+(``styling`` or ``layout``), ``old_id_dict``, ``new_id_dict`` and ``hash_dict``
+
+The function iterates through the key-value pairs of ``hash_dict`` and the contents
+of the list of elements; where the ``xml:id`` of both matches, the element is appended to
+the parent element. The hash and ``xml:id`` is then stored in ``new_id_dict``,
+where the hash is the ``key`` and the ``xml:id`` is the ``value``.
+
+In the final step, before emitting the document, the document as it is at this stage is passed
+through the :py:class:`ebu_tt_live.node.deduplicator.ReplaceStylesAndRegions` class.
```

Comment:
Worth saying this is a recursiveOperation, i.e. that the class is used to recursively fix the style and region attribute values.